revscoring/utilities/fetch_idioms.py

```diff
@@ -55,15 +56,49 @@ def fetch(output):
         continuation=True)
 
     idioms = []
+    first_term = ''
+    other_terms = []
     for doc in results:
         for page_doc in doc['query']['categorymembers']:
             # Some values take the form:
             # Category:English <variable>
             # For example: Category:English rhetorical questions
             # They are not idioms and should be filtered
             idiom = page_doc['title']
-            if 'Category:English' not in idiom:
-                idioms.append(idiom)
+        if re.match('Category:English|Citation:|Appendix:', idiom):
+            continue
+        # Remove one word idioms
+        if re.match(r'^[\w-]+$', idiom):
+            continue
+        # Remove two-worded idioms
+        if re.match(r'^[\w-]+ [\w-]+$', idiom):
+            continue
+        # Remove similes
+        if 'as a' in idiom:
+            continue
+        terms = idiom.split()
+        if terms[0] == first_term:
+            other_terms.append(" ".join(terms[1:]))
+        else:
+            if len(other_terms) < 2:
+                idiom_regex = '{} {}'.format(
+                    first_term, "".join(other_terms))
+            else:
+                idiom_regex = '{} ({})'.format(
+                    first_term, "|".join(other_terms))
+            if len(idiom_regex):
+                idioms.append(idiom_regex)
+            first_term = terms[0]
+            other_terms = [" ".join(terms[1:])]
+
+    if len(other_terms) < 2:
+        idiom_regex = '{} {}'.format(
+            first_term, "".join(other_terms))
+    else:
+        idiom_regex = '{} ({})'.format(
+            first_term, "|".join(other_terms))
+    if len(idiom_regex):

```

Comment:
How could this ever return false?  I think that `bool(len(" "))` is True.  That's the most *falsey* thing I could construct with `idiom_regex = '{} {}'.format(first_term, "".join(other_terms))`

Reply:
I missed that ... it used to be bool(len("")), but I added a space later on. I'll make changes.